### DISCUSSION TITLE
Can I get token/keycloakInstance in keycloakify?

### DISCUSSION THREAD

**Discussion by mikalai-repin** [keycloakify/keycloakify#382] [General]
Hi!
On the account page, I need to make a graphql request to get the "Terms of use" text.
The request must include a bearer token.
Can I somehow get a token/keycloakInstance in keycloakify?

**Comment by garronej** [comment]
I don't understand the question very well.  

What I can tell for sure is: yes, it's possible.  

The `useDowloadTerm` hook enables you to asyncly fetch your terms. Tou provide a `dowloadTermsMarkdon` that will be called when  needed. https://docs.keycloakify.dev/terms-and-conditions  

Even if this util wasn't provided, bear in mind that in Keycloakify you can do whatever you like. I provide a framwork to help you build your pages but at the end of the day, it's a regular react app, as long as you submit the proper form data to the Keycloak server you can do whatever you want client side.  

Now, a common thing you might want to avoid is hard coding an API url into your theme.  
If you want to prevent that you you have two options:  

- Pass a param in the URL while redirecting. This isn't the best approach since it expect the client app to provide the correct param in the URL. If your theme is only meant to tighly integrate with a given App it's no problem but otherwise you have option two.  
- You wan pass some environement variable that are set on the server runing Keycloak down to the react App application by using [extraThemeProperties](https://docs.keycloakify.dev/build-options#extrathemeproperties), see [this issue](https://github.com/keycloakify/keycloakify/issues/288) in particular.  

Hope it helps

**Reply by mikalai-repin** [reply]
@garronej Thanks for the reply. But I had something else in mind.
Now Keycloakify App uses static files (tos_fr.md, tos_en.md), from which the text for "Terms of use" is taken.
In my case, I need to make a request to the server, get the text and display it.
To make a request. I need to add a token to the headers
`headers: 
      { Authorization: Bearer eyJhbG...}`
      
I just can't find a way how I can get the token. After all, in fact, at the terms.ftl stage, the user has already been created in keycloak and theoretically he should already have a token.

**Reply by garronej** [reply]
Oh... I see, you want the access token. I uderstand the usecase, I don't know if Keycloak enables this though. The user is created but he's not enabled until he actually accepts the terms, if you have set it as required action anyway.  
Have you consider making requesting your terms unautenticated?  

Let me know if you find a solution

**Reply by mikalai-repin** [reply]
@garronej `Have you consider making requesting your auth unautenticated?` - this is one of the ways.
We will try to find some universal solution, but perhaps we will return to the fact that we will have to remove headers from requests.
Thank you for reply.
